### ISSUE TITLE
using vq-vae-2 for scoring probability of an image

### ISSUE THREAD

**Issue by zoharbarzelay** [rosinality/vq-vae-2-pytorch#36]
Thanks a lot for this repo!
quick question: once i train on your image set - can i somehow then use the model to estimate the probability that a new image is from the same distribution of the training set?

basically i want to use VAE to learn a prior model over images..

Thanks for anyone who can help with this!

Best,
Z.

**Comment by venergiac** [comment]
I reconstruct the image and I compare reconstructed with original using SSIM
![comparison](https://user-images.githubusercontent.com/9880255/87045951-caff4a80-c1f8-11ea-8928-2c058a9cd4fa.png)

**Comment by zoharbarzelay** [comment]
> 
> 
> I reconstruct the image and I compare reconstructed with original using SSIM
> ![comparison](https://user-images.githubusercontent.com/9880255/87045951-caff4a80-c1f8-11ea-8928-2c058a9cd4fa.png)

Thanks for your answer! could you plz provide some more details? for instance, what are the different colors in the graph?

**Comment by rosinality** [comment]
I think you can use prior (like PixelSNAIL) and reconstruction loss of the decoder to calculate likelihood.

**Comment by zoharbarzelay** [comment]
> 
> 
> I think you can use prior (like PixelSNAIL) and reconstruction loss of the decoder to calculate likelihood.

thanks for your answer! how would you suggest that i use PixelSNAIL?

**Comment by rosinality** [comment]
@zozobozo I think you can use probability estimates of PixelSNAIL that you can calculate by forward the code of the sample.
